src/app/wallets/get-transactions-for-wallet.ts

```diff
@@ -0,0 +1,59 @@
+import { RepositoryError } from "@domain/errors"
+import { OnChainError, MakeTxFilter, MakeTxDecoder } from "@domain/bitcoin/onchain"
+import { MakeWalletsRepository } from "@services/mongoose/wallets"
+import { MakeLedgerService } from "@services/ledger"
+import { MakeOnChainService } from "@services/lnd/onchain-service"
+import { toLiabilitiesAccountId, LedgerError } from "@domain/ledger"
+import { LOOK_BACK } from "@core/utils"
+import { ONCHAIN_MIN_CONFIRMATIONS } from "@config/app"
+import { WalletTransactionHistory } from "@domain/wallets"
+
+// TODO should be exposed via PriceSerivce / LiquidityProvider
+import { getCurrentPrice } from "@services/realtime-price"
+
+export const getTransactionsForWalletId = async ({
+  walletId,
+}: {
+  walletId: WalletId
+}) => {
+  const wallets = MakeWalletsRepository()
+  const wallet = await wallets.findById(walletId)
+  if (wallet instanceof RepositoryError) return wallet
+  return getTransactionsForWallet(wallet)
+}
+
+export const getTransactionsForWallet = async (
+  wallet: Wallet,
+): Promise<WalletTransaction[] | CoreError> => {
+  const ledger = MakeLedgerService()
+  const liabilitiesAccountId = toLiabilitiesAccountId(wallet.id)
+  const ledgerTransactions = await ledger.liabilityTransactions(liabilitiesAccountId)
+  if (ledgerTransactions instanceof LedgerError) return ledgerTransactions
+
+  const confirmedHistory = WalletTransactionHistory.fromLedger(ledgerTransactions)
+
+  const onChain = MakeOnChainService(MakeTxDecoder(process.env.NETWORK as BtcNetwork))
+  if (onChain instanceof OnChainError) {
+    return confirmedHistory.transactions
```

Comment:
may need to add an additional error to let the client know that not all the data has been returned